crawl-ref/source/menu.cc

```diff
@@ -1945,7 +2102,10 @@ void formatted_scroller::draw_index_item(int index, const MenuEntry *me) const
     if (me->data == nullptr)
         Menu::draw_index_item(index, me);
     else
+    {
+        static_cast<MenuDisplayText*>(mdisplay)->cgotoxy_for_index(index);
```

Comment:
The way in which`formatted_scroller` implies that mdisplay must be a `MenuDisplayText` is a bit convoluted right now; static_cast behavior is actually undefined when you cast a pointer to a subtype that doesn't match, so the fact that this compiles isn't a verification that the cast is possible.  I'd prefer to make that more direct somehow, perhaps adding an ASSERT to the formatted_scroller constructor (and something in a docstring).   (Right now it seems like you could make one crash in tiles by calling the constructor with `MF_USE_TWO_COLUMNS` as a flag, but the crash happens in the Menu constructor in a way that I don't think will be very obvious as to the problem.)  It would be even better to do this somehow as a compile-time `static_assert`, though it's not obvious to me how.  Alternatively, change the encapsulation so that casting isn't needed here.